src/MOI_wrapper.jl

```diff
@@ -0,0 +1,229 @@
+"""
+Constructs a Differentiable Optimizer model from a MOI Optimizer.
+Supports `forward` and `backward` methods for solving and differentiating the model respectectively.
+
+## Note 
+Currently supports differentiating linear and quadratic programs only.
+"""
+
+const VI = MOI.VariableIndex
+const CI = MOI.ConstraintIndex
+
+
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    optimizer::MOI.Bridges.LazyBridgeOptimizer
```

Comment:
Use `Optimizer{OT<:MOI.ModelLike}` and here do `optimizer::OT` so that it's a concrete type. Otherwise, the performance will be terrible. See https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-fields-with-abstract-containers-1

Reply:
fixed